src/constraint/SequentialSampleable.cpp

```diff
@@ -0,0 +1,177 @@
+#include "aikido/constraint/SequentialSampleable.hpp"
+
+#include <dart/common/Console.hpp>
+#include <dart/common/Memory.hpp>
+
+namespace aikido {
+namespace constraint {
+
+//==============================================================================
+class SequentialSampleGenerator : public SampleGenerator
+{
+public:
+  /// Sample Generator for Sequential Sampleable
+  /// \param[in] stateSpace Statespace the associated constraint operates in
+  /// \param[in] generators Sequence of generators associated with corresponding
+  /// sequence of sampleables
+  SequentialSampleGenerator(
+      statespace::StateSpacePtr stateSpace,
+      std::vector<std::unique_ptr<SampleGenerator>> generators);
```

Comment:
Should this be passed by reference? (https://app.codacy.com/app/personalrobotics/aikido/pullRequest?prid=1615713)

Reply:
Passing by _const_ reference is not allowed because `std::unique_ptr` is not copiable (see [this](https://www.reddit.com/r/cpp_questions/comments/7yvu89/stdvector_of_unique_ptr/)).

The codacy reports regarding smart pointers are somewhat inaccurate. :disappointed: